### DISCUSSION TITLE
Using global styles with SSR

### DISCUSSION THREAD

**Discussion by hardikmodi1** [vercel/next.js#40266] [Help]
We are hydrating the styles applied to components on server to client. in `_document.tsx` file. We also need global css, which is imported as `css` file in `_app.tsx` file. But facing issue in doing so. Our component styles on client are inserted above global css due to which some styles gets broken as global css will take precedence as it is below in the tree.

I have looked at how other libraries are doing this and fount out that they are providing some way to bring global css in react flow via component or function call, then use this in `_document.tsx`'s getInitialProps. So is there any other solution available to directly import global css and control the order of it being ordered in the DOM? 

Thanks!

**Comment by leerob** [comment]
Related: https://github.com/vercel/next.js/discussions/27953

We are looking to address this RFC ^ soon.
